Assignment/Assignment/SampleData.cs

```diff
@@ -1,30 +1,94 @@
 ﻿using System;
+using System.IO;
 using System.Collections.Generic;
+using System.Linq;
 
 namespace Assignment
 {
     public class SampleData : ISampleData
     {
-        // 1.
-        public IEnumerable<string> CsvRows => throw new NotImplementedException();
+        private List<string> _CsvRows;
 
-        // 2.
-        public IEnumerable<string> GetUniqueSortedListOfStatesGivenCsvRows() 
-            => throw new NotImplementedException();
+        public SampleData(string filePath)
+        {
 
-        // 3.
-        public string GetAggregateSortedListOfStatesUsingCsvRows()
-            => throw new NotImplementedException();
+            if (!File.Exists(filePath))//case if file does not exist
+            {
+                throw new FileNotFoundException("Error, file does not exist");
+            }
+            if (new FileInfo(filePath).Length == 0)//case if file passed in is empty
+            {
+                throw new ArgumentNullException("Error, file is empty");
+            }
 
-        // 4.
-        public IEnumerable<IPerson> People => throw new NotImplementedException();
+            try
+            {
+                _CsvRows = File.ReadAllLines(filePath).Skip(1).ToList();
+            }catch (Exception)
+            {
+                throw new ArgumentNullException("could not read from file");
+            }
+            
+            if (_CsvRows is null)
+            {
+                throw new ArgumentNullException("Error, reading lines from file went bad"); ;
+            }
+        }
 
-        // 5.
-        public IEnumerable<(string FirstName, string LastName)> FilterByEmailAddress(
-            Predicate<string> filter) => throw new NotImplementedException();
+        // 1.->Finished
+        public IEnumerable<string> CsvRows { get { return _CsvRows; } }
 
-        // 6.
+        // 2.->Finished
+        public IEnumerable<string> GetUniqueSortedListOfStatesGivenCsvRows()
+        {
+            List<string> states = new List<string>();
+            
+            states = CsvRows.Select(item => item.Split(",")).OrderBy(item => item[6]).Select(item => item[6]).Distinct().ToList();
+            states.Zip(states, (current, next) => string.Compare(current, next) <= 0).All(x => x);
+            return states;

```

Comment:
You have a lot of additional steps taking place that don't really need to be happening.
You should be able to solve the problem by simply doing the following:
```suggestion
            var states = CsvRows
                .Select(item => item.Split(",")[6])
                .OrderBy(item => item)
                .Distinct();
            return states;
```
You could even just do the return on one line if you wanted.

But it looks like you have two selects, you just need one select at the top and you do not need to convert to a list. You also do not need to use `.Zip`